working-docs/transient.md

```diff
@@ -0,0 +1,379 @@
+microarchitectural side effects that can be detected using timing or
+power analysis techniques. These techniques may allow an attacker to
+infer information about the operations that executed transiently. For
+example, the attacker may be able to infer program data that was
+accessed or used by those operations.
+
+### CWE-B: Transient Data Forwarding from an Operation that Triggers a Processor Event
+
+#### Description
+
+A processor event (for example, a fault or microcode assist) may allow
+incorrect data to be forwarded from the operation that triggered the
+event to operations that execute transiently.
+
+#### Extended Description
+
+Commodity processors may require assistance from microcode (or other
+processor design techniques) to execute certain operations. For example,
+processors that implement virtual memory may use microcode to perform
+virtual-to-physical address translation, check for faults, etc. Prior to
+or concurrent with the microcode assist, some processors may allow
+incorrect data to be forwarded from the operation that triggered the
+assist to operations that execute transiently and are flushed by the
+time the assist completes. These transient operations can affect
+observable microarchitectural state in a manner that could allow an
+attacker to infer program data, such as the incorrect data forwarded by
+the operation that triggered the assist.
+
+### CWE-C: Transient Execution Influenced by Shared Microarchitectural Predictor State
+
+#### Description
+
+Shared microarchitectural predictor state may allow code in one hardware
+domain to influence transient execution in another domain.
+
+#### Extended Description
+
+When hardware domains (for example, processes or privilege rings) share
+microarchitectural predictor state, code in one domain may be able to
+influence microarchitectural predictor behavior in another domain. If
+the predictor can cause transient execution, then shared predictor state
+may allow an attacker to influence transient execution in an
+attacker-chosen domain, and in a manner that could allow the attacker to
+infer program data from that domain.
+
+Many commodity processors have features which prevent microarchitectural
+predictions that occur before a domain transition from influencing
+predictions that occur after the domain transition. These features may
+be always-on, on by default, or may require opt-in from software.
+
+Some commodity processors may also share microarchitectural predictors
+(possibly including predictor state) between logical processors. These
+processors may have features that prevent microarchitectural predictions
+that occur on one logical processor from influencing predictions that
+occur on another logical processor. Similar to the features that prevent
+sharing across domain transitions, these features may be always-on, on
+by default, or may require opt-in from software.
+
+### CWE-D: Microarchitectural Predictor Causes Transient Execution
```

Comment:
Steven: The key issue here seems to be that the operations "affect observable microarchitectural state in a manner that could allow an attacker to infer program data" (said in the ext desc). If that's the key point, then it should be emphasized in the main desc